diff -ur msopenh264-1.1.1/src/msopenh264dec.cpp msopenh264-1.2.1/src/msopenh264dec.cpp
--- msopenh264-1.1.1/src/msopenh264dec.cpp	2015-09-23 15:14:22.000000000 +0200
+++ msopenh264-1.2.1/src/msopenh264dec.cpp	2016-10-19 17:32:52.000000000 +0200
@@ -14,6 +14,8 @@
 #include "msopenh264dec.h"
 #include "mediastreamer2/msticker.h"
 #include "ortp/b64.h"
+#include "wels/codec_ver.h"
+
 
 MSOpenH264Decoder::MSOpenH264Decoder(MSFilter *f)
 	: mFilter(f), mDecoder(0), mUnpacker(0), mSPS(0), mPPS(0), mYUVMsg(0),
@@ -53,11 +62,14 @@
 		mUnpacker=rfc3984_new();
 		if (mDecoder != 0) {
 			SDecodingParam params = { 0 };
+#if (OPENH264_MAJOR == 1) && (OPENH264_MINOR >=6)
+#else
 			params.eOutputColorFormat = videoFormatI420;
+#endif
 			params.uiTargetDqLayer = (unsigned char) -1;
 			params.eEcActiveIdc = ERROR_CON_FRAME_COPY_CROSS_IDR;
 			params.sVideoProperty.size = sizeof(params.sVideoProperty);
-			params.sVideoProperty.eVideoBsType = VIDEO_BITSTREAM_DEFAULT;
+			params.sVideoProperty.eVideoBsType = VIDEO_BITSTREAM_AVC;
 			long ret = mDecoder->Initialize(&params);
 			if (ret != 0) {
 				ms_error("OpenH264 decoder: Failed to initialize: %li", ret);
diff -ur msopenh264-1.1.1/src/msopenh264enc.cpp msopenh264-1.2.1/src/msopenh264enc.cpp
--- msopenh264-1.1.1/src/msopenh264enc.cpp	2015-09-23 15:14:22.000000000 +0200
+++ msopenh264-1.2.1/src/msopenh264enc.cpp	2017-07-20 15:56:32.000000000 +0200
@@ -14,7 +14,7 @@
 #include "msopenh264enc.h"
 #include "mediastreamer2/msticker.h"
 #include "wels/codec_app_def.h"
-
+#include "wels/codec_ver.h"
 
 //static const int RC_MARGIN = 10000; // bits per sec
 #if MSOPENH264_DEBUG
@@ -114,9 +112,15 @@
 			params.sSpatialLayers[0].fFrameRate = mVConf.fps;
 			params.sSpatialLayers[0].iSpatialBitrate = targetBitrate;
 			params.sSpatialLayers[0].iMaxSpatialBitrate = maxBitrate;
+
+#if (OPENH264_MAJOR == 1) && (OPENH264_MINOR >=6)
+			params.sSpatialLayers[0].sSliceArgument.uiSliceMode = SM_SIZELIMITED_SLICE;
+			params.sSpatialLayers[0].sSliceArgument.uiSliceSizeConstraint = ms_factory_get_payload_max_size(mFilter->factory);
+#else
 			params.sSpatialLayers[0].sSliceCfg.uiSliceMode = SM_DYN_SLICE;
 			params.sSpatialLayers[0].sSliceCfg.sSliceArgument.uiSliceSizeConstraint = ms_factory_get_payload_max_size(mFilter->factory);
-
+#endif
+			//params.sSpatialLayers[0].sSliceCfg.uiSliceMode = SM_AUTO_SLICE;
 			ret = mEncoder->InitializeExt(&params);
 			if (ret != 0) {
 				ms_error("OpenH264 encoder: Failed to initialize: %d", ret);
--- msopenh264-1.1.1/src/msopenh264enc.h~	2015-09-23 15:14:22.000000000 +0200
+++ msopenh264-1.1.1/src/msopenh264enc.h	2018-09-29 13:18:20.730187249 +0200
@@ -21,7 +21,6 @@ Foundation, Inc., 59 Temple Place - Suit
 #include "mediastreamer2/msfilter.h"
 #include "mediastreamer2/msvideo.h"
 #include "mediastreamer2/rfc3984.h"
-#include "mediastreamer2/videostarter.h"
 #include "wels/codec_api.h"
 #include "msopenh264.h"
 
